django/contrib/admin/utils.py

```diff
@@ -362,6 +368,10 @@ def label_for_field(name, model, model_admin=None, return_attr=False):
                     label = pretty_name(attr.__name__)
             else:
                 label = pretty_name(name)
+    except FieldIsAForeignKeyColumnName:
+        label = pretty_name(name)
```

Comment:
Yes. `label_for_field("site_id", Article)` which is included in this PR tests this branch. 

I have taken care of your other comments as well. Hope this is good to go.